tools/clang/test/HLSLFileCheck/pix/TraceRayInlineDebug.hlsl

```diff
@@ -0,0 +1,30 @@
+// RUN: %dxc -T vs_6_5 -Od -E main %s | %opt -S -dxil-annotate-with-virtual-regs -hlsl-dxil-debug-instrumentation | FileCheck %s
+ 
+// CHECK: [[RAYQUERY0:%.*]] = call i32 @dx.op.allocateRayQuery
+// CHECK: [[RAYQUERY1:%.*]] = call i32 @dx.op.allocateRayQuery
+
+// CHECK: [[PHIQUERY:%.*]] = phi i32 [ [[RAYQUERY0:%.*]], {{.*}} ], [ [[RAYQUERY0:%.*]], {{.*}} ]
+
+// The debug instrumentation should NOT try to store this phi value: it's not an i32! (It's an opaque handle).
```

Comment:
If the DXIL validator let you do this, I think validation is deficient for RayQuery handle.  I'm not surprised, but an issue should be filed about that.

Reply:
#6312

Reply:
In DXIL it is just an i32 value though, is that something that validator should/could check?

Reply:
That's what I filed #6312 for...